### PR TITLE
Extend Probe Helper to cover ApiServerSource

### PR DESCRIPTION
**Pull request by stuart-xyz**
Extends Probe Helper functionality to cover ApiServerSource, a built-in Knative source.

## Proposed Changes

* Add ApiServerSource HTTP endpoint to Probe Helper, for use by testing infrastructure.
* Add Probe Helper ApiServerSource test.

### CODE REVIEW

File: test/test_images/probe_helper/utils/k8s_client.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import "k8s.io/client-go/kubernetes"
+
+type K8sClient struct {
```
Comment (tommyreddad) [inline test/test_images/probe_helper/utils/k8s_client.go:21]: Do we need this type (& this file)? Why can't we just use kubernetes.Interface directly?

Comment (stuart-xyz) [reply test/test_images/probe_helper/utils/k8s_client.go:21]: We don't -- I originally created this to solve a problem that's no longer relevant. Will remove.

File: test/test_images/probe_helper/probe/helper_test.go [hunk 2]
```diff
@@ -175,6 +193,55 @@ func runTestCloudAuditLogsSource(ctx context.Context, group *errgroup.Group, pub
...
+				return nil
+			case req := <-gotRequest:
+				bodyBytes, err := ioutil.ReadAll(req.Body)
+				if err != nil {
+					logging.FromContext(ctx).Warnf("Failed to read request body in test ApiServerSource, %v", err)
+				}
+				body := string(bodyBytes)
+				method := req.Method
+				url := req.URL.String()
+				logging.FromContext(ctx).Warnf("DEBUG: %s", url)
```
[the remark this answers is not in the document]
Comment (stuart-xyz) [reply test/test_images/probe_helper/probe/helper_test.go]: Nope, just removed it.

**Comment by tommyreddad** [comment]
/lgtm
/approve

Great work! 🎉

**Pull request merged**